### PR TITLE
Sort agents by success rate in end game summary

### PR DESCRIPTION
**Pull request by Crotalus**
## About the PR
Sort the agents by success rate in the end game summary
## Why / Balance
Fame and glory to the robust

**Changelog**
:cl:
- tweak: Sort agents by success rate in end game summary

### CODE REVIEW

File: Content.Server/Objectives/ObjectivesSystem.cs [hunk 1]
```diff
@@ -114,25 +116,25 @@ private void AddSummary(ref string result, string agent, List<EntityUid> minds)
...
+            var totalObjectives = 0;
+            var agentSummary = Loc.GetString("objectives-with-objectives", ("custody", custody), ("title", title), ("agent", agent));
 
             foreach (var objectiveGroup in objectives.GroupBy(o => Comp<ObjectiveComponent>(o).Issuer))
             {
                 //TO DO:
                 //check for the right group here. Getting the target issuer is easy: objectiveGroup.Key
                 //It should be compared to the type of the group's issuer.
-                result += "\n" + Loc.GetString($"objective-issuer-{objectiveGroup.Key}");
+                agentSummary += "\n" + Loc.GetString($"objective-issuer-{objectiveGroup.Key}");
```
Comment (metalgearsloth) [inline Content.Server/Objectives/ObjectivesSystem.cs:138]: Use stringbuilder.

File: Content.Server/Objectives/ObjectivesSystem.cs [hunk 2]
```diff
@@ -161,7 +166,17 @@ private void AddSummary(ref string result, string agent, List<EntityUid> minds)
...
+            var successRate = totalObjectives > 0 ? (float) completedObjectives / totalObjectives : 0f;
+            agentSummaries.Add((agentSummary, successRate, completedObjectives));
         }
+
+        agentSummaries = agentSummaries.OrderByDescending(x => x.successRate)
+                                       .ThenByDescending(x => x.completedObjectives)
+                                       .ToList();
+
+        foreach (var (summary, _, _) in agentSummaries)
+            result += "\n" + summary;
```
Comment (metalgearsloth) [inline Content.Server/Objectives/ObjectivesSystem.cs:179]: Same here use a stringbuilder, there's also no reason to ToList.

**Review by metalgearsloth — DISMISSED** [review]

File: Content.Server/Objectives/ObjectivesSystem.cs (hunk 1 above)
Comment (Crotalus) [reply Content.Server/Objectives/ObjectivesSystem.cs]: Fixed, changed result parameter from ref string to StringBuilder as well

File: Content.Server/Objectives/ObjectivesSystem.cs (hunk 2 above)
Comment (Crotalus) [reply Content.Server/Objectives/ObjectivesSystem.cs]: Fixed

**Pull request merged**